widget/toolbar.go

```diff
@@ -17,27 +17,51 @@ type ToolbarItem interface {
 type ToolbarAction struct {
 	Icon        fyne.Resource
 	OnActivated func() `json:"-"`
+	button      *Button
```

Comment:
I'm not sure how I feel about embedding button directly here. What I might prefer is adding a `Disabled  bool` here, and creating a private `toolbarActionButton` type here that extends `widget.Button`, holds a reference to the parent `ToolbarAction`, and in its overridden `Refresh` function, syncs the state from the `ToolbarAction` to the button, and then call `BaseWidget.Refresh()`.

Then, in `ToolbarObject()`, create and return a new `toolbarActionButton` that has the reference to `t`.

Reply:
Hi, thanks for reviewing the PR so quickly.

I think I got your point about the private toolbarActionButton, although I am not entirely sure why you prefer this option above using the button type directly as a member of the ToolbarAction type. Could you please elaborate more on this decision?

Reply:
The Fyne APIs are designed to allow changing properties through setter functions (eg `Disable()`) or properties (`wid.Disabled = true`). With embedding the button directly the Disable property will not be a part of the API for ToolbarAction. Also we try to separate the state from the renderer - see the separation between `Widget` and `WidgetRenderer` in the codebase. The button is what's used to render the toolbar action, not really an intrinsic property of the action.

Reply:
> The button is what's used to render the toolbar action, not really an intrinsic property of the action.
Got it, thank you.

I implemented the suggested `toolbarActionButton`. I intentionally used the `Disabled` property, since this is also what the `MenuItem` in `menu.go` uses, and I wanted to stay consistent here.
What I somehow dislike about the property approach is, that it requires an explicit refresh, which would not be the case if one would use the function-API (`Enable()`/`Disable()`) instead of the property (`Disabled`), because the refresh could be hidden in the function-API.

Let me know what you think.